openedx/core/djangolib/translation_utils.py

```diff
@@ -0,0 +1,22 @@
+from django.utils.translation import ugettext as _, override
+from django.utils.formats import dateformat, get_format
+
+
+def translate_date(date, language, date_format='DATE_FORMAT'):
+    """
+    Converts the provided date object into a string, while translating
+    its value for the given language.  Both the format of the date
+    as well as its values (i.e., name of the Month) are translated.
+
+    For example:
+        date = datetime.datetime(2017, 12, 23)
+        date_in_spanish = translate_date(date, 'es')
+        assert date_in_spanish = '12 de Deciembre de 2017'
+    """
+    with override(language):
+        formatted_date = dateformat.format(
+            date,
+            get_format(date_format, lang=language, use_l10n=True),
+        )
+        translated_date = _('{formatted_date}').format(formatted_date=formatted_date)
```

Comment:
I assume this is the core of the fix? Or is it the language override?

I don't understand what this line is doing...

Reply:
Yes, it's the combination of the language override and the call to `ugettext` here that is the crux of the fix.